src/compose.js

```diff
@@ -12,15 +12,18 @@ var reverse = require('./reverse');
  * @memberOf R
  * @since v0.1.0
  * @category Function
- * @sig ((y -> z), (x -> y), ..., (o -> p), ((a, b, ..., n) -> o)) -> ((a, b, ..., n) -> z)
- * @param {...Function} functions
+ * @sig ((x -> z), ..., (b -> c), ((a1, a2, ...) -> b), (a1, a2, ...)) -> ((a1, a2, ...) -> z)
```

Comment:
Note that we're not following any formal notation, just hacking together whatever seem to be appropriate extensions to Hindley-Milner type signature descriptions to capture Javascript's differences.  I wrote a [lengthy article][wi] on how we're using them in Ramda.

>  Personally I found it clearer to show that only the first function was variadic by using a1, a2, ...

I certainly understand the temptation, but what bothers me is that now we have to represent an arbitrary type, not just the single-lowercase-letter tokens but also any of those followed by a number.  Would `a3` be fine on its own, or only in conjunction with `a1` and `a2`?  It just seems to make the rules more complex.  I don't totally abhor them, but I'm hoping we can find a way to avoid them.

  [wi]: https://github.com/ramda/ramda/wiki/Type-Signatures